components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -516,28 +520,23 @@ public boolean isReady() {
         return ready;
     }
 
-    Properties getKafkaProps() {
+    private Properties getKafkaProps() {
         return kafkaProps;
     }
 
-    String getClientId() {
-        return clientId;
-    }
-
-    Exception getLastError() {
-        return lastError;
-    }
-
-    boolean isTerminated() {
+    private boolean isTerminated() {
         return terminated;
     }
 
-    boolean isRecoverable() {
+    private boolean isRecoverable() {
         return (pollExceptionStrategy.canContinue() || isReconnect()) && isKafkaConsumerRunnable();
     }
 
-    long getCurrentRecoveryInterval() {
-        return currentBackoffInterval;
+    // concurrent access happens here
+    public TaskHealthState healthState() {
```

Comment:
private package instead of public as it seems to be an implementation detail?

Reply:
+1 from me, but let's leave it like this for now. I am considering moving the whole set of health check code into a separate package (or, at least, within each of `consumer` and `producer` packages).